cmake/Embed.cmake

```diff
@@ -21,17 +21,21 @@
 # OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 # THE SOFTWARE.
 #####################################################################################
-find_program(EMBED_LD ld)
-find_program(EMBED_OBJCOPY objcopy)
 
-option(EMBED_USE_LD "Use ld to embed data files" OFF)
+if(WIN32)
+    option(EMBED_USE_RESOURCES "Use data file embedding to binary" ON)
+else()
+    option(EMBED_USE_RESOURCES "Use data file embedding to binary" OFF)
+endif()
+
+if(NOT WIN32 AND EMBED_USE_RESOURCES)
+    find_program(EMBED_LD ld)
```

Comment:
Why are we searching for `ld` when `EMBED_USE_RESOURCES` is enabled on linux? This should be an error.

Reply:
The technique with LD on Linux is the same as RC on Windows. I changed `EMBED_USE_LD` to `EMBED_USE_RESOURCES` because, in both cases, the literature refers to both techniques as embedding resource files in program binary. Here is the example: https://atwillys.de/content/cc/embedding-resource-files-in-a-c-plus-plus-program-binary-on-linux-unix/?lang=en. I can easily find tonnes more. We only need one option to control resources or `*.cpp` files. It makes no sense to have both and expose `EMBED_USE_LD` on Windows and `EMBOED_USE_RC` on Linux.